### PR TITLE
enable application to run

### PR DESCRIPTION
**Pull request by Rex2002**
this version of the program now runs, when not enough shapes are present. 
Still needs to be tested with shapes actually present that could be recognized

### CODE REVIEW

**Comment by Rex2002** [comment]
1. color of shape drawings
2. somehow manage to place the drawn shapes in relation to the image-view, not the whole frame
currently the shapes x and y position is considered to be relative to the whole frames x and y which leads to differences between the drawn shapes and the shapes on the frame, when the frame has not the same size as the image
3. definition of playfield
for the marker classification the playfield is averaged to be a rectangle based on the recognized field_markers, since this avoids calculating the play-times of the shapes in a non-uniform way between the "upper" bars and the "lower" bars. 
Since the playfield is drawn by connecting the playfield markers, this may lead to unclarity, especially when the playmarker (indicating the position in the bar) is also drawn
4. how should the playmarker be passed?
@MalteRichert your opinion?

**Comment by MalteRichert** [comment]
@Rex2002 addressing your comments:
1. Color: Should be determined by Shape Type, right? So that e.g. FieldMarkers are black, SoundMarkers are blue etc. Will prioritize that in styling
2. will try to set shape position in relation to ShapePane and scale ShapePane with Image. That should do the trick
3. Imo that leaves two options: either the UI recalculates the exact PlayField (risky and inefficient) or the playField is somehow passed from ShapeProcessing. That could just be an array with four tuples of x,y-coordinates. Can we do that?
4. PlayMarker could be just an ordinary Setting with the relative position within the loop. It could be its own thing within UIMessage too, but I don't see any reason for that at this time.

**Comment by Rex2002** [comment]
1. noice
2. very noice
3. yes, thats possible, currently the playfield is stored in an array containing the x and y position of the upper left corner, the width, height and whether it is somehow similar to a rectangle (stored as 0/1), but the x and y's of the corners can be calculated from that
4. indeed :ok_hand:

File: src/main/java/de/dhbw/Main.java [hunk 1]
```diff
@@ -60,7 +65,20 @@ public static void runController(){
             markerRecognizer.detectShapes();
             shapeProcessor.processShapes(markerRecognizer.getShapes(), frame.width(), frame.height());
             midiAdapter.tickMidi(clock.currentBeat, shapeProcessor.getSoundMatrix(), settings);
+            EventQueues.toUI.offer(new UIMessage(markerRecognizer.getShapes()));
             // TODO add sending shapes to UI for display
```
Comment (MalteRichert) [inline src/main/java/de/dhbw/Main.java]: This `TODO` is done, right?

Comment (Rex2002) [reply src/main/java/de/dhbw/Main.java]: yes :+1:

File: src/main/java/de/dhbw/ui/VideoScene.java [hunk 2]
```diff
@@ -58,68 +67,67 @@ private void handleQueue() {
...
+        else {
+            System.out.println("VideoScene: Can't draw PlayField because none is present.");
+        }
+
         MatOfByte buffer = new MatOfByte();
         Imgcodecs.imencode(".png", frame, buffer);
-        Image image = new Image( new ByteArrayInputStream(buffer.toArray() ) );
+        Image image = new Image( new ByteArrayInputStream( buffer.toArray() ) );
 
+        currentFrame.setImage(null);
```
Comment (MalteRichert) [inline src/main/java/de/dhbw/ui/VideoScene.java]: @Rex2002 what does this do? Isn't it redundant given the next line?

**Comment by MalteRichert** [comment]
@Rex2002 could this be ready for a merge? I still need to do some stuff before the initial UI skeleton is done but I think that can be a new branch.

**Review by MalteRichert — APPROVED** [review]

**Pull request merged**